src/sbvr-api/permissions.ts

```diff
@@ -1089,15 +1092,14 @@ const getReqPermissions = (req: PermissionReq, odataBinds: ODataBinds = []) =>
 	Promise.join(
 		getGuestPermissions(),
 		Promise.try(() => {
-			if (
-				req.apiKey != null &&
-				req.apiKey.permissions != null &&
-				req.apiKey.permissions.length > 0
-			) {
-				return getApiKeyActorId(req.apiKey.key);
+			// TODO: Remove this extra actor ID lookup making actor non-optional and updating open-balena-api.
+			if (req.apiKey != null && req.apiKey.actor == null) {
```

Comment:
And on that logic this also needs to be
```suggestion
			if (req.apiKey != null && req.apiKey.actor == null && req.apiKey.permissions != null && req.apiKey.permissions.length > 0) {
```

Reply:
Asked above: I don't understand why we need this permissions length check.